xskillscore/core/probabilistic.py

```diff
@@ -488,32 +549,55 @@ def rps(
     """Calculate Ranked Probability Score.
 
      .. math::
-        RPS(p, k) = 1/M \\sum_{m=1}^{M}
-        [(\\sum_{k=1}^{m} p_k) - (\\sum_{k=1}^{m} o_k)]^{2}
+        RPS = \\sum_{m=1}^{M}[(\\sum_{k=1}^{m} y_k) - (\\sum_{k=1}^{m} o_k)]^{2}
+
+    where ``y`` and ``o`` are forecast and observation probabilities in ``M``
+    categories.
+
+     .. note::
+        Takes the sum over all categories as in Weigel et al. 2007 and not the mean as
+        in https://www.cawcr.gov.au/projects/verification/verif_web_page.html#RPS.
+        Therefore RPS has no upper boundary.
 
     Parameters
     ----------
     observations : xarray.Dataset or xarray.DataArray
-        The observations or set of observations of the event.
-        Data should be boolean or logical \
-        (True or 1 for event occurance, False or 0 for non-occurance).
+        The observations of the event.
+        Further requirements are specified based on ``category_edges``.
     forecasts : xarray.Dataset or xarray.DataArray
-        The forecast likelihoods of the event.
-        If ``fair==False``, forecasts should be between 0 and 1 without a dimension
-        ``member_dim`` or should be boolean (True,False) or binary (0, 1) containing a
-        member dimension (probabilities will be internally calculated by
-        ``.mean(member_dim))``. If ``fair==True``, forecasts must be boolean
-        (True,False) or binary (0, 1) containing dimension ``member_dim``.
-    category_edges : array_like
-        Category bin edges used to compute the CDFs. Similar to np.histogram, \
-        all but the last (righthand-most) bin include the left edge and exclude \
-        the right edge. The last bin includes both edges.
+        The forecast of the event.
+        Further requirements are specified based on ``category_edges``.
+    category_edges : array_like, xr.Dataset, xr.DataArray, None
+        Category edges used to compute the CDFs. Similar to np.histogram, all but the
+        last (righthand-most) bin include the left edge and exclude the right edge. The
+        last bin includes both edges.
+        These ``category_edge`` must span the full range of the observations and
+        forecasts distribution. Forecasts, observations and category_edge are expected
+        in absolute units or probabilities consistently.
+        ``category_edges`` decides how xs.rps interprets forecasts and observations.
```

Comment:
actually the rightmost edges is not bound at the top by the algorithm. but I understand that the category_edges have to span the full range, e.g. no observations or forecasts are allowed outside the category_edges. because the cumulative dist. functions have to end up at 1 both, which is why RPS can actually be calculated without the last one. unsure how explicit this should be explained here and whether the code should be adapted

Reply:
I think we need to be explicit about this. If I understand correctly, the current explanation about being similar to np.histogram is not correct. What about:

"Bins include the left edge and are defined to always include the full range of the observations and forecasts distributions. That is, the left-most bin contains data between -infinity and the first element of `category_edges`, and the right-most bin contains data between the last element of `category edges` and +infinity."